athena-jdbc/src/main/java/com/amazonaws/connectors/athena/jdbc/saphana/SAPHANAMetadataHandler.java

```diff
@@ -0,0 +1,196 @@
+    {
+        super(databaseConnectionConfig, new GenericJdbcConnectionFactory(databaseConnectionConfig, JDBC_PROPERTIES));
+    }
+
+    @VisibleForTesting
+    protected SAPHANAMetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig, final AWSSecretsManager secretsManager,
+            AmazonAthena athena, final JdbcConnectionFactory jdbcConnectionFactory)
+    {
+        super(databaseConnectionConfig, secretsManager, athena, jdbcConnectionFactory);
+    }
+
+    @Override
+    public Schema getPartitionSchema(final String catalogName)
+    {
+        
+        SchemaBuilder schemaBuilder = SchemaBuilder.newBuilder()
+               .addField(BLOCK_PARTITION_COLUMN_NAME, Types.MinorType.VARCHAR.getType());
+                
+        return schemaBuilder.build();
+    }
+
+    @Override
+    public void getPartitions(final BlockWriter blockWriter, final GetTableLayoutRequest getTableLayoutRequest, QueryStatusChecker queryStatusChecker)
+    {
+        LOGGER.info("{}: Schema {}, table {}", getTableLayoutRequest.getQueryId(), getTableLayoutRequest.getTableName().getSchemaName(),
+                getTableLayoutRequest.getTableName().getTableName());
+        try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider())) {
+            final String escape = connection.getMetaData().getSearchStringEscape();
+
+            List<String> parameters = Arrays.asList(getTableLayoutRequest.getTableName().getTableName().toUpperCase(), getTableLayoutRequest.getTableName().getSchemaName().toUpperCase());
+            try (PreparedStatement preparedStatement = new PreparedStatementBuilder().withConnection(connection).withQuery(GET_PARTITIONS_QUERY).withParameters(parameters).build();
+                    ResultSet resultSet = preparedStatement.executeQuery()) {
+                // Return a single partition if no partitions defined
+                if (!resultSet.next()) {
+                    blockWriter.writeRows((Block block, int rowNum) -> {
+                        block.setValue(BLOCK_PARTITION_COLUMN_NAME, rowNum, ALL_PARTITIONS);
+                        LOGGER.info("Adding partition {}", ALL_PARTITIONS);
+                        //we wrote 1 row so we return 1
+                        return 1;
+                    });
+                }
+                else {
+                    do {
+                        final String partitionName = resultSet.getString(PARTITION_COLUMN_NAME);
+
+                        // 1. Returns all partitions of table, we are not supporting constraints push down to filter partitions.
+                        // 2. This API is not paginated, we could use order by and limit clause with offsets here.
+                        blockWriter.writeRows((Block block, int rowNum) -> {
+                            block.setValue(BLOCK_PARTITION_COLUMN_NAME, rowNum, partitionName);
+                            LOGGER.info("Adding partition {}", partitionName);
```

Comment:
Is this logging necessary? How verbose will this be?

Reply:
Not required. Was mainly for debugging. Removed